src/features/PowerShellNotebooks.ts

```diff
@@ -0,0 +1,220 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { CommentType } from '../settings';
+import { IFeature, LanguageClient } from '../feature';
+import { EvaluateRequestType } from './Console';
+import Settings = require("../settings");
+
+export class PowerShellNotebooksFeature implements vscode.NotebookContentProvider, vscode.NotebookKernel, IFeature {
+
+    private readonly showNotebookModeCommand: vscode.Disposable;
+    private readonly hideNotebookModeCommand: vscode.Disposable;
+    private languageClient: LanguageClient;
+
+    private _onDidChangeNotebook = new vscode.EventEmitter<vscode.NotebookDocumentEditEvent>();
+    onDidChangeNotebook: vscode.Event<vscode.NotebookDocumentEditEvent> = this._onDidChangeNotebook.event;
+    kernel?: vscode.NotebookKernel;
+
+    constructor() {
+        this.kernel = this;
+        this.showNotebookModeCommand = vscode.commands.registerCommand("PowerShell.ShowNotebookMode", async () => {
+            const uri = vscode.window.activeTextEditor.document.uri;
+            await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+            await vscode.commands.executeCommand("vscode.openWith", uri, "PowerShellNotebookMode");
+        });
+
+        this.hideNotebookModeCommand = vscode.commands.registerCommand("PowerShell.HideNotebookMode", async () => {
+            const uri = vscode.notebook.activeNotebookEditor.document.uri;
+            await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+            await vscode.commands.executeCommand("vscode.openWith", uri, "default");
+        });
+    }
+
+    label: string = 'PowerShell';
+    preloads?: vscode.Uri[];
+
+    async openNotebook(uri: vscode.Uri, context: vscode.NotebookDocumentOpenContext): Promise<vscode.NotebookData> {
+        // load from backup if needed.
+        const actualUri = context.backupId ? vscode.Uri.parse(context.backupId) : uri;
+
+        const data = (await vscode.workspace.fs.readFile(actualUri)).toString();
+        const lines = data.split(/\r|\n|\r\n/g);
+
+        const notebookData: vscode.NotebookData = {
+            languages: ["powershell"],
+            cells: [],
+            metadata: {}
+        }
+
+        let curr: string[] = [];
+        let cellKind: vscode.CellKind | undefined;
+        let insideBlockComment: boolean = false;
+
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < lines.length; i++) {
```

Comment:
Maybe add a comment about what this loop is doing (or possibly turn it into a method). It's not clear from the logic

Reply:
added a few extra comments.